### PR TITLE
Add new gtk-rs-core 0.20 / gtk4-rs 0.9 release

### PR DESCRIPTION
**Pull request by sdroege**

### CODE REVIEW

File: _posts/2024-07-17-new-release.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+libadwaita `0.7` and other related crates got a new release compatible with
+the new gtk4-rs and gtk-rs-core releases and their own set of changes.
+
+### gtk-rs-core
+
+#### New syntax for `glib::clone!` and `glib::closure!`
+
+blablablabla
+
+#### TBD
```
Comment (sdroege) [inline _posts/2024-07-17-new-release.md:26]: Update to GLib/etc 2.82 APIs

File: _posts/2024-07-17-new-release.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+blablablabla
+
+### gtk4-rs
```
Comment (sdroege) [inline _posts/2024-07-17-new-release.md:30]: Update to GTK4 4.16 APIs

File: _posts/2024-07-17-new-release.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+ * [glib: Add unsafe `Value::into\_send\_value()`](https://github.com/gtk-rs/gtk-rs-core/pull/1413)
+ * [glib: Improve `ValueArray` API, add tests and assertions for invalid …](https://github.com/gtk-rs/gtk-rs-core/pull/1411)
+ * [glib: Fix `MatchInfo::next()` handling of returning `FALSE`](https://github.com/gtk-rs/gtk-rs-core/pull/1410)
+ * [glib/functions: add compute\_checksum\_for\_string](https://github.com/gtk-rs/gtk-rs-core/pull/1406)
+ * [glib: Add bindings for `g\_value\_set\_static\_string()`](https://github.com/gtk-rs/gtk-rs-core/pull/1400)
+ * [docs: Fix broken links / cleanup README](https://github.com/gtk-rs/gtk-rs-core/pull/1395)
+ * [Fix clippy warnings](https://github.com/gtk-rs/gtk-rs-core/pull/1393)
+ * [glib: Implement Sync for ThreadGuard](https://github.com/gtk-rs/gtk-rs-core/pull/1388)
+ * [glib: Only implement Send on JoinHandle if the result is Send](https://github.com/gtk-rs/gtk-rs-core/pull/1387)
+ * [glib: Decouple ObjectInterface impl from interface class struct](https://github.com/gtk-rs/gtk-rs-core/pull/1384)
```
Comment (sdroege) [inline _posts/2024-07-17-new-release.md:107]: Maybe will write a sentence about this and show code

File: _posts/2024-07-17-new-release.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+ * [glib: Add missing Send bound to the output type of the `spawn\_from\_within()` future](https://github.com/gtk-rs/gtk-rs-core/pull/1383)
+ * [glib-macros: Refactor parsing code of object\_subclass/object\_interface](https://github.com/gtk-rs/gtk-rs-core/pull/1379)
+ * [examples: Add example for custom class structs and virtual methods](https://github.com/gtk-rs/gtk-rs-core/pull/1378)
+ * [ObjectBuilder: add property\_if(), property\_if\_some(), property\_from\_iter() ...](https://github.com/gtk-rs/gtk-rs-core/pull/1377)
+ * [glib: Don't use `g\_object\_list\_properties()` for setting properties](https://github.com/gtk-rs/gtk-rs-core/pull/1376)
+ * [glib: Use a reference to a pointer of correct mutability for from\_gli…](https://github.com/gtk-rs/gtk-rs-core/pull/1375)
```
Comment (sdroege) [inline _posts/2024-07-17-new-release.md:113]: Maybe will write a sentence about this and show code

File: _posts/2024-07-17-new-release.md [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+#### New syntax for `glib::clone!` and `glib::closure!`
+
+The syntax for the `clone!` and `closure!` macros was updated to look more
+like valid Rust code, and as a side effect it is also handled correctly by
+`cargo fmt`, rust-analyzer and other tooling now.
+
+The old syntax is still supported but will give a deprecation warning.
+
+To get an idea of the change, what previously looked like
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:30]: Missing `:`.

File: _posts/2024-07-17-new-release.md [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+
+```rust
+clone!(@strong self as obj, @weak v => @default-return false, move |x| {
+    println!("obj: {}, v: {}, x: {}", obj, v, x);
+    true
+})
+```
+
+would now look like this
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:39]: Missing `:`.

File: _posts/2024-07-17-new-release.md [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+
+New GLib and GIO 2.82 APIs are supported with this release. GLib 2.56 is still
+the minimum version supported by the bindings.
+
+#### Trait re-organization for defining new GObject interfaces
+
+The traits for defining new GObject interfaces were slightly re-organized to
+make them more similar with the ones for defining new GObjects.
+
+Previously one would write
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:68]: Missing `:`.

File: _posts/2024-07-17-new-release.md [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+    parent: glib::gobject_ffi::GTypeInterface,
+}
+
+#[glib::object_interface]
+unsafe impl ObjectInterface for MyStaticInterface {
+    const NAME: &'static str = "MyStaticInterface";
+}
+```
+
+This would now become
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:83]: Missing `:`.

File: _posts/2024-07-17-new-release.md [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+New GTK 4.16 APIs are supported with this release. GTK 4.0 is still the
+minimum version supported by the bindings.
+
+### Changes
+
+For the interested ones, here is the list of the merged pull requests:
+
+[gtk4-rs](https://github.com/gtk-rs/gtk4-rs):
+
+ * [Regenerate with new gir/gir-files](https://github.com/gtk-rs/gtk4-rs/pull/1792)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:145]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+ * [Update to new clone! macro syntax](https://github.com/gtk-rs/gtk4-rs/pull/1773)
+ * [book: Fix typo](https://github.com/gtk-rs/gtk4-rs/pull/1769)
+ * [Remove unnecessary upcast from examples/squeezer\_bin/main.rs](https://github.com/gtk-rs/gtk4-rs/pull/1763)
+ * [typos: Ignore versions.txt file](https://github.com/gtk-rs/gtk4-rs/pull/1762)
+ * [examples/squeezer\_bin: Mark properties as writable](https://github.com/gtk-rs/gtk4-rs/pull/1761)
+ * [Fix `SqueezerBin::size\_allocate()` in example](https://github.com/gtk-rs/gtk4-rs/pull/1760)
+ * [Regen bindings with new gir {g,s}etter annotations](https://github.com/gtk-rs/gtk4-rs/pull/1759)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:152]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+ * [Stop renaming ffi crates](https://github.com/gtk-rs/gtk4-rs/pull/1758)
+ * [custom\_orientable: Fix interface property override](https://github.com/gtk-rs/gtk4-rs/pull/1755)
+ * [gtk: Implement Downgrade for TemplateChild&lt;T&gt;](https://github.com/gtk-rs/gtk4-rs/pull/1750)
+ * [Update list\_widgets.md](https://github.com/gtk-rs/gtk4-rs/pull/1742)
+ * [examples: Support GL &gt;= 3.1 and GLES &gt;= 3.0 in the glium example](https://github.com/gtk-rs/gtk4-rs/pull/1733)
+ * [Fix nightly clippy warnings](https://github.com/gtk-rs/gtk4-rs/pull/1730)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:158]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+ * [Update link for Cogitri/Health to World/Health](https://github.com/gtk-rs/gtk4-rs/pull/1725)
+ * [Simplify reading file contents to a String](https://github.com/gtk-rs/gtk4-rs/pull/1719)
+ * [Update dependencies](https://github.com/gtk-rs/gtk4-rs/pull/1717)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:161]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 13]
```diff
@@ -0,0 +1,294 @@
...
+ * [macros: Drop anyhow dependency](https://github.com/gtk-rs/gtk4-rs/pull/1708)
+ * [Fix typo in todo\_1.md](https://github.com/gtk-rs/gtk4-rs/pull/1707)
+ * [gtk4: Manually implement `GraphicsOffload` constructor for now](https://github.com/gtk-rs/gtk4-rs/pull/1705)
+ * [gtk4: Require GDK 4.14 when enabling the `v4\_14` feature](https://github.com/gtk-rs/gtk4-rs/pull/1704)
+ * [macros: Drop macro-proc-error and upgrade syn to 2.0](https://github.com/gtk-rs/gtk4-rs/pull/1688)
+ * [dockerfile: Update libadwaita to 1.5](https://github.com/gtk-rs/gtk4-rs/pull/1687)
+ * [book: Update dependencies](https://github.com/gtk-rs/gtk4-rs/pull/1686)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:168]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 14]
```diff
@@ -0,0 +1,294 @@
...
+ * [Regenerate with latest gir](https://github.com/gtk-rs/gtk4-rs/pull/1678)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:169]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 15]
```diff
@@ -0,0 +1,294 @@
...
+ * [docs: fix `Path` setting on windows](https://github.com/gtk-rs/gtk4-rs/pull/1675)
+ * [Correctly handle `NULL` `GError\*\*` out parameters](https://github.com/gtk-rs/gtk4-rs/pull/1672)
+ * [Fix nightly clippy warnings](https://github.com/gtk-rs/gtk4-rs/pull/1670)
+ * [Replace simple `impl Debug` with derived `Debug` in tokio example](https://github.com/gtk-rs/gtk4-rs/pull/1663)
+ * [Simplify library configuration step for Windows](https://github.com/gtk-rs/gtk4-rs/pull/1644)
+ * [Regenerate with latest gir-files](https://github.com/gtk-rs/gtk4-rs/pull/1637)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:175]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 16]
```diff
@@ -0,0 +1,294 @@
...
+ * [cargo-deny: Remove libloading dependencies](https://github.com/gtk-rs/gtk4-rs/pull/1633)
+
+[gtk-rs-core](https://github.com/gtk-rs/gtk-rs-core):
+
+ * [ci: Fix docs release tag handling](https://github.com/gtk-rs/gtk-rs-core/pull/1457)
+ * [docs: Run on our container image](https://github.com/gtk-rs/gtk-rs-core/pull/1455)
+ * [gio: Add a method to get a stream of incoming connections to SocketListener](https://github.com/gtk-rs/gtk-rs-core/pull/1454)
+ * [Update gir files](https://github.com/gtk-rs/gtk-rs-core/pull/1453)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:183]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 17]
```diff
@@ -0,0 +1,294 @@
...
+ * [glib: Add support for registering GTypes with name conflicts](https://github.com/gtk-rs/gtk-rs-core/pull/1451)
+ * [glib: Make `TypeData` struct fields private](https://github.com/gtk-rs/gtk-rs-core/pull/1449)
+ * [Update gir files & regenerate](https://github.com/gtk-rs/gtk-rs-core/pull/1448)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:186]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 18]
```diff
@@ -0,0 +1,294 @@
...
+ * [glib-macros: Don't produce unnecessary braces in `clone!(async move {…](https://github.com/gtk-rs/gtk-rs-core/pull/1443)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:187]: Missing the end of the sentence.

File: _posts/2024-07-17-new-release.md [hunk 19]
```diff
@@ -0,0 +1,294 @@
...
+ * [Update to system-deps 7](https://github.com/gtk-rs/gtk-rs-core/pull/1440)
+ * [glib-macros: Fix unit return in `closure!()` macro](https://github.com/gtk-rs/gtk-rs-core/pull/1438)
+ * [strv: add From implementation from a String array](https://github.com/gtk-rs/gtk-rs-core/pull/1432)
+ * [Derive TransparentPtrType trait for Boxed](https://github.com/gtk-rs/gtk-rs-core/pull/1431)
+ * [gio: Properly export Win32InputStream / Win32OutputStream traits](https://github.com/gtk-rs/gtk-rs-core/pull/1429)
+ * [ci: bump gvsbuild](https://github.com/gtk-rs/gtk-rs-core/pull/1427)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:193]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 20]
```diff
@@ -0,0 +1,294 @@
...
+ * [glib: Add bindings for `g\_value\_set\_static\_string()`](https://github.com/gtk-rs/gtk-rs-core/pull/1400)
+ * [docs: Fix broken links / cleanup README](https://github.com/gtk-rs/gtk-rs-core/pull/1395)
+ * [Fix clippy warnings](https://github.com/gtk-rs/gtk-rs-core/pull/1393)
+ * [glib: Implement Sync for ThreadGuard](https://github.com/gtk-rs/gtk-rs-core/pull/1388)
+ * [glib: Only implement Send on JoinHandle if the result is Send](https://github.com/gtk-rs/gtk-rs-core/pull/1387)
+ * [glib: Decouple ObjectInterface impl from interface class struct](https://github.com/gtk-rs/gtk-rs-core/pull/1384)
+ * [glib: Add missing Send bound to the output type of the `spawn\_from\_within()` future](https://github.com/gtk-rs/gtk-rs-core/pull/1383)
+ * [glib-macros: Refactor parsing code of object\_subclass/object\_interface](https://github.com/gtk-rs/gtk-rs-core/pull/1379)
+ * [examples: Add example for custom class structs and virtual methods](https://github.com/gtk-rs/gtk-rs-core/pull/1378)
+ * [ObjectBuilder: add property\_if(), property\_if\_some(), property\_from\_iter() ...](https://github.com/gtk-rs/gtk-rs-core/pull/1377)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:214]: Missing the end of the sentence.

File: _posts/2024-07-17-new-release.md [hunk 21]
```diff
@@ -0,0 +1,294 @@
...
+ * [glib: Don't use `g\_object\_list\_properties()` for setting properties](https://github.com/gtk-rs/gtk-rs-core/pull/1376)
+ * [glib: Use a reference to a pointer of correct mutability for from\_gli…](https://github.com/gtk-rs/gtk-rs-core/pull/1375)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:216]: Missing the end of the sentence.

File: _posts/2024-07-17-new-release.md [hunk 22]
```diff
@@ -0,0 +1,294 @@
...
+ * [glib: Re-add and rename manual Win32 API additions](https://github.com/gtk-rs/gtk-rs-core/pull/1372)
+ * [glib-macros/properties: Allow structs with no properties](https://github.com/gtk-rs/gtk-rs-core/pull/1370)
+ * [glib::wrapper: Add docs for impls generated by the wrapper macro](https://github.com/gtk-rs/gtk-rs-core/pull/1369)
+ * [ci: Enable glib-sys / gobject-sys tests](https://github.com/gtk-rs/gtk-rs-core/pull/1368)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:220]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 23]
```diff
@@ -0,0 +1,294 @@
...
+ * [glib-sys: Add missing includes in `manual.h`](https://github.com/gtk-rs/gtk-rs-core/pull/1361)
+ * [glib-sys: fix struct size mismatches](https://github.com/gtk-rs/gtk-rs-core/pull/1360)
+ * [glib: Freeze property notifications while setting multiple properties](https://github.com/gtk-rs/gtk-rs-core/pull/1355)
+ * [glib-macros: Improve error message when `Properties` struct doesn't have at least one `#\[property(…)\]`](https://github.com/gtk-rs/gtk-rs-core/pull/1352)
+ * [Update gir / gir-files and update for `time\_t`](https://github.com/gtk-rs/gtk-rs-core/pull/1349)
+ * [Fix nightly clippy warnings](https://github.com/gtk-rs/gtk-rs-core/pull/1348)
+ * [docs: `construct` attribute for `glib::Properties`](https://github.com/gtk-rs/gtk-rs-core/pull/1344)
+ * [glib: fix UB in VariantStrIter::impl\_get](https://github.com/gtk-rs/gtk-rs-core/pull/1343)
+ * [Fix glib compiletest expected output for Rust 1.77](https://github.com/gtk-rs/gtk-rs-core/pull/1342)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:229]: Can be removed.

File: _posts/2024-07-17-new-release.md [hunk 24]
```diff
@@ -0,0 +1,294 @@
...
+ * [glib: Don't misuse `slice::get\_unchecked()`](https://github.com/gtk-rs/gtk-rs-core/pull/1337)
+ * [gio: correctly free argument list items](https://github.com/gtk-rs/gtk-rs-core/pull/1331)
+ * [glib: Optimize string collation bindings a bit](https://github.com/gtk-rs/gtk-rs-core/pull/1329)
+ * [glib: Drop the main context future return value sender on finalize](https://github.com/gtk-rs/gtk-rs-core/pull/1328)
+ * [pango: add some missing AttrInt constructors.](https://github.com/gtk-rs/gtk-rs-core/pull/1325)
+ * [Regenerate with latest gir files](https://github.com/gtk-rs/gtk-rs-core/pull/1324)
```
Comment (GuillaumeGomez) [inline _posts/2024-07-17-new-release.md:235]: Can be removed.

**Pull request merged**